services/QuillLMS/app/queries/student_dashboard_metrics.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+class StudentDashboardMetrics
+  include DashboardMetrics
+
+  def initialize(user, classroom_id)
+    @user = user
+    @classroom_id = classroom_id
+  end
+
+  def run
+    {
+       day: metrics_from_start_date(today),
+       week: metrics_from_start_date(last_sunday),
+       month: metrics_from_start_date(today.beginning_of_month),
+       year: metrics_from_start_date(last_july_first)
+     }
+  end
+
+  def metrics_from_start_date(start_date)
+    completed_sessions_for_timeframe = completed_sessions.where("completed_at >= ?", start_date)
+    {
+      activities_completed: completed_sessions_for_timeframe.count,
+      timespent: completed_sessions_for_timeframe.pluck(:timespent).sum { |num| num || 0 }
```

Comment:
I think this is more efficient since it avoids explicit nil checks:
`timespent: completed_sessions_for_timeframe.pluck(:timespent).compact.sum`